src/main/java/com/mathworks/ci/RunMatlabCommandBuilder.java

```diff
@@ -0,0 +1,137 @@
+        }
+
+        /*
+         * This is to identify which project type in jenkins this should be applicable.(non-Javadoc)
+         * 
+         * @see hudson.tasks.BuildStepDescriptor#isApplicable(java.lang.Class)
+         * 
+         * if it returns true then this build step will be applicable for all project type.
+         */
+        @Override
+        public boolean isApplicable(
+                @SuppressWarnings("rawtypes") Class<? extends AbstractProject> jobtype) {
+            return true;
+        }
+    }
+
+    @Override
+    public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace,
+            @Nonnull Launcher launcher, @Nonnull TaskListener listener)
+            throws InterruptedException, IOException {
+
+        // Set the environment variable specific to the this build
+        setEnv(build.getEnvironment(listener));
+
+        // Invoke MATLAB command and transfer output to standard
+        // Output Console
+
+        buildResult = execMatlabCommand(workspace, launcher, listener, getEnv());
+
+        if (buildResult != 0) {
+            build.setResult(Result.FAILURE);
+        }
+    }
+
+    private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher,
+            TaskListener listener, EnvVars envVars) throws IOException, InterruptedException {
+        ProcStarter matlabLauncher;
+        try {
+            // Get mMatlabroot set in wrapper.
+            String matlabRoot = envVars.get("matlabroot");
+            CommandConstructUtil cmdUtils = new CommandConstructUtil(launcher, matlabRoot);
+            matlabLauncher = launcher.launch().pwd(workspace).envs(envVars);
+            FilePath targetWorkspace = new FilePath(launcher.getChannel(), workspace.getRemote());
+            if(launcher.isUnix()) {
+                matlabLauncher = launcher.launch().pwd(workspace).envs(envVars).cmds("/bin/bash","-c","./run_matlab_command.sh "+cmdUtils.constructCommandForRunCommand(getCommand())).stdout(listener);
+                //Copy runner .sh for linux platform in workspace.
+                cmdUtils.copyMatlabScratchFileInWorkspace(MatlabBuilderConstants.SHELL_RUNNER_SCRIPT, "Builder.matlab.runner.script.target.file.linux.name", targetWorkspace);
+            }else {
+                launcher = launcher.decorateByPrefix("cmd.exe","/C");
+                matlabLauncher = launcher.launch().pwd(workspace).envs(envVars).cmds("run_matlab_command.bat","\""+cmdUtils.constructCommandForRunCommand(getCommand())+"\"").stdout(listener);
+                //Copy runner.bat for Windows platform in workspace.
+                cmdUtils.copyMatlabScratchFileInWorkspace(MatlabBuilderConstants.BAT_RUNNER_SCRIPT, "Builder.matlab.runner.script.target.file.windows.name", targetWorkspace);
+            }
+        } catch (Exception e) {
+            listener.getLogger().println(e.getMessage());
+            return 1;
+        }
+        return matlabLauncher.join();

```

Comment:
Consider refactoring. It looks like a large portion of this code block is repeated in `RunMatlabCommandBuilder` and `RunMatlabTestsBuilder`.

Reply:
Done. added new Interface with default method implementations and both builders are now implementing this interface.